client_plugin/utils/refcount/refcnt.go

```diff
@@ -447,10 +449,15 @@ func (r *RefCountsMap) syncMountsWithRefCounters(d drivers.VolumeDriver) {
 			}
 		} else {
 			if cnt.count == 0 {
-				// volume unmounted AND refcount 0.  We should NEVER get here
-				// since unmounted and recount==0 volumes should have no record
-				// in the map. Something went seriously wrong in the code.
-				log.WithFields(f).Panic("Internal failure: record should not exist. ")
+				// This volume may be attached to the VM but not mounted as no containers
+				// are using it. A detach is attempted and the entry under the mount root is
+				// removed.
+				err := d.DetachVolume(vol)
+				if err != nil {
+					log.Warning("Failed to detach %s - manual recovery may be needed", vol)
+				}
+				fs.Rmdir(strings.Join([]string{mountRoot, vol}, "/"))
```

Comment:
If this op wasn't around till date, how were the entries from mountRoot removed?

In such a case, there will be entries under mountRoot even for the volumes that have been successfully detached.
Due to updateRefMap() reading this entries, they ll be populated in refMap.
An attempt to detach them would fail (since they were already detached before).
We ll end up flagging false warning for such cases through line 457.